front/poke/temporal/activities.ts

```diff
@@ -40,3 +85,424 @@ export async function scrubDataSourceActivity({
     );
   }
 }
+
+export async function isWorkflowDeletableActivity({
+  workspaceId,
+}: {
+  workspaceId: string;
+}) {
+  const auth = await Authenticator.internalBuilderForWorkspace(workspaceId);
+  const workspace = auth.workspace();
+  if (!workspace) {
+    return false;
+  }
+
+  // Workspace must have no data sources.
+  const dataSources = await DataSource.findAll({
+    where: {
+      workspaceId: workspace.id,
+      visibility: {
+        [Op.or]: ["public", "private", "unlisted"],
```

Comment:
Insn't that basically all of them ? Do we want this filter at all ?

Reply:
We use this filter everywhere and I 🐒 pasted it 🙈